File: src/java/magmac/app/lang/java/node/Annotation.java
package magmac.app.lang.java.node;

import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;
import magmac.app.lang.java.Deserializers;

public record Annotation(String value) {
    public static CompileResult<Annotation> deserialize(Node node) {
        return Deserializers.deserialize(node)
                .withString("value")
                .complete(Annotation::new);
    }
}

File: src/web/magmac/app/lang/node/Annotation.ts
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Deserializers } from "../../../../magmac/app/lang/Deserializers";
export class Annotation {
	deserialize(node : Node) : CompileResult<Annotation>;
}
